### PR TITLE
[SDK] Feature/use voucher preflight check refacto

### PR DESCRIPTION
**Pull request by Le-Caignec**
- [x] Import Reordering: Imports are reordered for better readability and maintainability.
- [x] Preflight Check Separation: Preflight checks are split into two separate functions to improve code organization and clarity.
- [x] Modularization of Contract Instances: The getContract instance is moved to a dedicated file, as it's made for getPocoContract, getSharingContract ... for better modularity.
- [x] Smart Contract Read Functions: Read functions for smart contracts are moved to their respective file (xxx.read.ts), ensuring better structure and separation of concerns.
- [x] Top-Level Preflight Check Placement: Preflight checks are moved to the top of the functions for improved early error handling and code structure.
- [x] Promise.all Utilization: Promise.all is utilized for improved asynchronous operation performance where applicable.

### CODE REVIEW

**Comment by PierreJeanjacquot** [comment]
@Le-Caignec once the build issue (imports, lint, format) are fixed there is an issue with `consumeProtectedData` (loading voucher when `useVoucher` is not specified)

tests without vouchers are falling:
```
  ● dataProtector.consumeProtectedData() › When whitelist contract does not have registered delivery app › should throw error


    expect(received).rejects.toThrow(expected)


    Expected substring: "This whitelist contract does not have registered this app: 0x1cb7d4f3ffa203f211e57357d759321c6ce49921."

    Received message:   "No Voucher found for address 0x03561e5A3ab3a41AE14E4c71FA51153cDDEB9176"


          22 | ): Promise<IVoucher> => {

          23 |   const { signer } = await iexec.config.resolveContractsClient();

        > 24 |   const voucherInfo = await iexec.voucher.showUserVoucher(userAddress);

             |                       ^

          25 |   return new Contract(voucherInfo.address, VOUCHER_ABI).connect(

          26 |     signer

          27 |   ) as IVoucher;


      at showUserVoucher (node_modules/iexec/dist/esm/common/voucher/voucher.js:31:13)

      at getVoucherContract (src/lib/dataProtectorSharing/smartContract/getVoucherContract.ts:24:23)

      at consumeProtectedData (src/lib/dataProtectorSharing/consumeProtectedData.ts:105:27)

      at Object.<anonymous> (tests/e2e/dataProtectorSharing/consumeProtectedData.test.ts:97:9)

      at Object.toThrow (node_modules/expect/build/index.js:218:22)

      at Object.toThrow (tests/e2e/dataProtectorSharing/consumeProtectedData.test.ts:104:19)
```

**Comment by Le-Caignec** [comment]
fix Commit: d3a99a598733fb4637f5ac140bf690a92d4022ba

File: packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts [hunk 1]
```diff
@@ -392,3 +400,74 @@ export const onlyAccountWithMinimumBalance = ({
...
+
+    if (
+      Number(accountAllowance) === 0 ||
+      Number(accountAllowance) < workerpoolPrice
+    ) {
+      throw new Error(
+        `Voucher balance is insufficient to sponsor workerpool. Please approve an additional ${missingAmount} for voucher usage.`
+      );
+    }
+  }
```
Comment (PierreJeanjacquot) [inline packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts:472]: do the maths with bigints instead of casting to number

**Review by PierreJeanjacquot — APPROVED** [review]
see comments

File: packages/sdk/package-lock.json
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply packages/sdk/package-lock.json:1]: fix Commit: a6a05e0f79c45a52844d7eefcc550a300ea50b60

File: packages/sdk/src/lib/types/pocoTypes.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
 export type AccountDetails = {
   balance: bigint;
-  sharingContractAllowance: bigint;
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply packages/sdk/src/lib/types/pocoTypes.ts:3]: When useVoucher is set to true the the allowance is not for the Dataprotector Sharing contract but for the voucherContract.
This is why i rename it

[the remark this answers is not in the document]
Comment (PierreJeanjacquot) [reply packages/sdk/src/lib/types/pocoTypes.ts:3]: ok just a comment about point of view

File: packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts (hunk 1 above)
Comment (Le-Caignec) [reply packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts]: fix :Commit: 7c16db7d76f4fa4003163f82e720ef98c9a9a5a3

**Pull request merged**